packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1561,4 +1616,30 @@ class CarouselController extends ScrollController {
     carouselPosition.itemExtent = _carouselState!._itemExtent;
     carouselPosition.consumeMaxWeight = _carouselState!._consumeMaxWeight;
   }
+
+  /// Animates the carousel to a specific item index.
+  ///
+  /// [index]: The index of the item to animate to.
+  /// [duration]: Duration of the animation.
+  /// [curve]: The animation curve to use.
+  Future<void> animateToItem(
+    int index, {
+    Duration duration = const Duration(milliseconds: 300),
+    Curve curve = Curves.ease,
+  }) async {
+    // Ensure the state is attached
+    if (_carouselState == null) {
+      throw StateError('CarouselController is not attached to a CarouselView.');
+    }
+
+    // Calculate the target offset using the item's index and extent
+    final double targetOffset = index * _carouselState!._itemExtent;
```

Comment:
`_itemExtent` can be null when using `CarouselView.weighted`

Reply:
Yes, and the position of the actually item can be dependent on weights, as they change in response to scrolling. This case will need to be addressed and tested.